content/docs/user-guide/how-to/add-deps-or-outs-to-a-stage.md

```diff
@@ -0,0 +1,55 @@
+# How to Add Dependencies or Outputs
+
+We have executed a stage, but later notice that some of the files/directories it
+uses as dependencies or creates as outputs are missing from `dvc.yaml`...
+
+To add files or directories as <abbr>dependencies</abbr> or <abbr>outputs</abbr>
+to an existing stage without re-executing it (which can be expensive and is
```

Comment:
I would still simplify it:

To add files or directories ... edit the `dvc.yaml` (alternatively do `dvc run --no-exec` if you still have this command around).

Now, to update the `dvc.lock` either run `dvc repro` or `dvc commit` to "force" it in case outputs already exist and you don't want to run anything again.

-----

It's better to push people editing file manually. `dvc run` is complicated, people don't have it around sometimes, etc.

Reply:
> to update the dvc.lock either run dvc repro

But the scenario is "add deps/outs to an existing stage **without re-executing it**". I simplified it along the lines of your suggestion for now, but skipped the parts that contradict that.